### PR TITLE
Expand and Unexpand top nodes in Treetable

### PR DESCRIPTION
**Pull request by carolynqu**
Fixing: top nodes in treetable cannot be expanded or unexpanded when clicked on. 
bug

### CODE REVIEW

**Review by elliette — APPROVED** [review]

**Comment by carolynqu** [comment]
Solution Options:
- Make it so that when changing DiffTreeTable type we also call initState and expand the roots then and remove the expand roots from didUpdateWidget
- Remove expandRoots from didUpdateWidget and instead place them in a location where DiffTreeTable is being modified.
- Investigate why root.isExpanded is changing from false to true when clicking on it. This causes expandRoot to expand the roots just after it has been collapsed by toggleNode which is called first.

File: packages/devtools_app/lib/src/shared/table.dart [hunk 1]
```diff
@@ -417,6 +417,7 @@ class TreeTableState<T extends TreeNode<T>> extends State<TreeTable<T>>
   @override
   void initState() {
     super.initState();
+
```
Comment (elliette) [inline packages/devtools_app/lib/src/shared/table.dart:420]: Let's remove newline

File: packages/devtools_app/lib/src/shared/table.dart [hunk 2]
```diff
@@ -453,24 +454,26 @@ class TreeTableState<T extends TreeNode<T>> extends State<TreeTable<T>>
...
 
-    if (widget.sortColumn != oldWidget.sortColumn ||
+    if (widget.sortColumn.title != oldWidget.sortColumn.title ||
         widget.sortDirection != oldWidget.sortDirection ||
         !collectionEquals(widget.dataRoots, oldWidget.dataRoots)) {
       _initData();
     }
     rootsExpanded =
         List.generate(dataRoots.length, (index) => dataRoots[index].isExpanded);
+
```
Comment (elliette) [inline packages/devtools_app/lib/src/shared/table.dart:464]: Remove new lien

File: packages/devtools_app/lib/src/shared/table.dart [hunk 3]
```diff
@@ -453,24 +454,26 @@ class TreeTableState<T extends TreeNode<T>> extends State<TreeTable<T>>
...
   void _initData() {
     dataRoots = List.generate(widget.dataRoots.length, (index) {
       final root = widget.dataRoots[index];
-      if (widget.autoExpandRoots) {
+      if (widget.autoExpandRoots && !root.isExpanded) {
         root.expand();
       }
       return root;
     });
+
```
Comment (elliette) [inline packages/devtools_app/lib/src/shared/table.dart:476]: Remove new line

**Review by elliette — APPROVED** [review]
LGTM once new lines are removed!

**Pull request merged**